Northstar.CustomServers/mod/scripts/vscripts/gamemodes/_gamemode_fd.nut

```diff
@@ -1231,7 +1249,36 @@ void function OnHarvesterDamaged( entity harvester, var damageInfo )
 
 	float shieldPercent = ( ( harvester.GetShieldHealth().tofloat() / harvester.GetShieldHealthMax() ) * 100 )
 	if ( shieldPercent < 100 && !file.harvesterShieldDown )
-		PlayFactionDialogueToTeam( "fd_baseShieldTakingDmg", TEAM_MILITIA )
+	{
+		switch(damageSourceID)
+		{
+			case eDamageSourceId.titanEmpField:
+			PlayFactionDialogueToTeam( "fd_nagKillTitanEMP", TEAM_MILITIA )
+			break
+		
+			case eDamageSourceId.damagedef_stalker_powersupply_explosion_small:
+			PlayFactionDialogueToTeam( "fd_nagKillStalkers", TEAM_MILITIA )
+			break
+			
+			case eDamageSourceId.damagedef_stalker_powersupply_explosion_large:
+			PlayFactionDialogueToTeam( "fd_nagKillStalkers", TEAM_MILITIA )
+			break
+			
+			case eDamageSourceId.mp_weapon_rspn101: //Not sure how i would do a Grunt check, so i went for R-201 damage type which is the weapon they use
+			PlayFactionDialogueToTeam( "fd_nagKillInfantry", TEAM_MILITIA )
+			break
+			
+			case eDamageSourceId.mp_titanweapon_rocketeer_rocketstream:
+			PlayFactionDialogueToTeam( "fd_nagKillTitansMortar", TEAM_MILITIA )
+			break
+			
+			default:
+			PlayFactionDialogueToTeam( "fd_baseShieldTakingDmg", TEAM_MILITIA )
+			break
+			
+			//Mortar Spectres are still pending, but i'm not sure what damage type they inflict on the Harvester yet
```

Comment:
maybe `int attackerTypeID = FD_GetAITypeID_ByString( attacker.GetTargetName() )` is much better